src/handler/mod.rs

```diff
@@ -1067,22 +1044,11 @@ impl Handler {
         response: Response,
     ) {
         // Find a matching request, if any
-        if let Some(mut request_call) = self.active_requests.remove(&node_address) {
-            let id = match request_call.id() {
-                HandlerReqId::Internal(id) | HandlerReqId::External(id) => id,
-            };
-            if id != &response.id {
-                trace!(
-                    "Received an RPC Response to an unknown request. Likely late response. {}",
-                    node_address
-                );
-                // add the request back and reset the timer
-                self.active_requests.insert(node_address, request_call);
-                return;
-            }
-
+        if let Ok(mut request_call) = self
+            .active_requests
+            .remove_request(&node_address, &response.id)
+        {
```

Comment:
Why remove this logic?

Reply:
It's no longer needed, due to the change in `ActiveRequests`. The exception that the removed logic was catching (`response.id` doesn't match request id in `active_requests_mapping` hashmap) is now handled inside `ActiveRequests::remove_request(..)` and will return `None`, which is handled in the `else` statement.